lib/config/rule-validator.js

```diff
@@ -36,7 +36,12 @@ function throwRuleNotFoundError({ pluginName, ruleName }, config) {
     const ruleId = pluginName === "@" ? ruleName : `${pluginName}/${ruleName}`;
 
     const errorMessageHeader = `Key "rules": Key "${ruleId}"`;
-    let errorMessage = `${errorMessageHeader}: Could not find plugin "${pluginName}".`;
+    const error = new TypeError(`${errorMessageHeader}: Could not find plugin "${pluginName}" in configuration.`);
```

Comment:
We don't need to create the error here. If we wait until the end (like it is in HEAD) then we can avoid a lot of this extra code.

Just set the `errorMessage` here.